Backup App migration

jacob-nv opened the pull request:
<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for Web. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.

Please set the following labels:

- Set label "Status:Needs-Review" for review or "Status:In-Progress" in case the PR still has open tasks
- Set label "Category:*" where it fits best
- Assignment: assign to self
- Reviewers: pick at least one
-->

## Description
<!--- Describe your changes in detail -->
Moved back up app, moved all the dependencies from web-app-files.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes https://github.com/owncloud/web/issues/9267


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

dschmidt commented on packages/web-pkg/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts:
```diff
@@ -0,0 +1,119 @@
+export const useResourcesViewDefaults = <T, TT, TU extends any[]>(
+  options: ResourcesViewDefaultsOptions<TT, TU> = {}
+): ResourcesViewDefaultsResult<T, TT, TU> => {
+  const loadResourcesTask = options.loadResourcesTask || folderService.getTask()
```
I don't think it's a really good idea to move the whole FolderLoaderService and all of the loaders into web-pkg at this point.

I would suggest to make `options.loadResourcesTask` of type: ` Task<TT, TU> || () =>  Task<TT, TU>`
and then pass it in everywhere where useResourcesViewDefaults is used: `{ loadResourcesTask: () => folderService.getTask() } `

jacob-nv replied:
Looked at it, the main issue is that the folderService is located in web-app-files. Which shouldn't be reachable from web-app-backups

dschmidt replied:
Yes, that's why I said you could inject a getTask function...

dschmidt replied:
My main issue is that there are so many route checks in the loaders, so they are actually pretty files app specific...

jacob-nv replied:
I might have explained poorly.
`packages/web-app-backups/src/components/BackupsSection.vue
` line: 141

uses `useResourcesViewDefaults`, which then should get the injected `{ loadResourcesTask: () => folderService.getTask() }`, the issue here is that for `folderService` to be injected, it would have to have access to _web-app-files_, which goes against the goal of this issue, no?

dschmidt replied:
Aaah, the app itself needs the loaders? Grmpfmhl

JammingBen replied:
Okay so turns out the backups app only uses the `useResourcesViewDefaults` composable to get `selectedResourcesIds`. Hence we suggest to leave the `useResourcesViewDefaults` and all the loaders in the files-app and only move the `useSelectedResources` composable to web-pkg.

jacob-nv replied:
Done, but I could have missed unnecessary files that could be removed as well.

jacob-nv commented on packages/web-pkg/src/composables/scrollTo/useScrollTo.ts:
Well, there's actually different id for topbar. So I could modify it to pass the id with default being 'files-app-bar' and specifically for admin settings pass 'admin-settings-app-bar'.

dschmidt replied:
or no default at all and require it

JammingBen replied:
> or no default at all and require it

Yes that sounds like the best solution IMO. The composable shouldn't make any assumption about the selector here.

jacob-nv replied:
Done. 
Then it could be replaced everywhere no?

JammingBen replied:
Yes, all other usages of this composable should consume it from `web-pkg` now.

jacob-nv replied:
Done

jacob-nv commented on packages/web-pkg/src/components/AppBar/AppBar.vue:
```diff
@@ -0,0 +1,342 @@
+          </template>
+        </oc-breadcrumb>
+        <portal-target v-if="showMobileNav" name="app.runtime.mobile.nav" />
+        <shares-navigation v-if="hasSharesNavigation" />
```
Done

jacob-nv commented on packages/web-pkg/src/components/FilesViewWrapper.vue:
Done

JammingBen commented on packages/web-pkg/src/components/AppBar/AppBar.vue:
```diff
@@ -0,0 +1,341 @@
+          </template>
+        </oc-breadcrumb>
+        <portal-target v-if="showMobileNav" name="app.runtime.mobile.nav" />
+        <slot v-if="hasSharesNavigation" name="navigation" />
```
I think we can remove `hasSharesNavigation` as property and replace it with a `computed` that checks, if the slot `navigation` is given.

JammingBen commented on packages/web-pkg/src/helpers/resources.ts:
Hmm maybe it makes sense to have that file in `web-client`, alongside with the other files that include our `build...` methods. @dschmidt What do you think?

JammingBen commented on packages/web-pkg/src/helpers/resource/ancestorMetaData.ts:
Please remove that file, we already have the types in `packages/web-pkg/src/helpers/resource/ancestorMetaData.ts`. Some imports need to be adjusted then of course 🙂

JammingBen reviewed (commented):
Getting there, nice 🥳 Some open tasks:

* Remove the moved files from the files-app
* Adjust all affected imports
* Move test `*.spec.ts` files accordingly
* Linter seems to be unhappy
* At the very end: remove the `web-app-backup` package from this PR. After it has been merged, we need to publish a new `web-pkg` package and then create a PR with the `web-app-backup` package in the CERN repo.

jacob-nv commented:
@JammingBen 
- moved all the relevant tests to web-pkg
- removed all the moved files from web-app-files
- changed imports to web-pkg

Some unit tests are still not happy.
Not really sure what to do with _web-app-files/tests/unit/components/AppBar/CreateAndUpload.spec.ts_
Getting error of _Cannot read properties of undefined (reading 'mockImplementation')_
